GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MultithreadedReadWriteTests.cs

```diff
@@ -14,8 +14,48 @@ namespace GVFS.FunctionalTests.Tests.EnlistmentPerFixture
     [Category(Categories.Mac.M1)]
     public class MultithreadedReadWriteTests : TestsWithEnlistmentPerFixture
     {
-        [TestCase]
-        public void CanReadUnhydratedFileInParallelWithoutTearing()
+        [TestCase, Order(1)]
+        public void CanReadVirtualFileInParallel()
+        {
+            // Note: This test MUST go first, or else it needs to ensure that it is reading a unique path compared to the
+            // other tests in this class. That applies to every directory in the path, as well as the leaf file name.
+            // Otherwise, this test loses most of its value because there will be no races occurring on creating the
+            // placeholder directories, enumerating them, and then creating a placeholder file and hydrating it.
+            
+            string fileName = Path.Combine("GVFS", "GVFS.FunctionalTests", "Tests", "LongRunningEnlistment", "GitMoveRenameTests.cs");
+            string virtualPath = this.Enlistment.GetVirtualPathTo(fileName);
+
+            Exception readException = null;
+
+            Thread[] threads = new Thread[32];
+            for (int i = 0; i < threads.Length; ++i)
+            {
+                int myIndex = i;
+                threads[i] = new Thread(() =>
+                {
+                    try
+                    {
+                        FileSystemRunner.DefaultRunner.ReadAllText(virtualPath).ShouldBeNonEmpty();
+                    }
+                    catch (Exception e)
+                    {
+                        readException = e;

```

Comment:
Does the number of failed reads provide any interesting signal?

Reply:
I wondered about that too, but I'm not sure that it would. The number needs to be 0, always. And if it's ever non-zero, it signals that there's at least one race, somewhere, but gives no more signal than that. So I think this is a binary test, unless you can think of more we can get out of it?

Reply:
> So I think this is a binary test, unless you can think of more we can get out of it?

@sanoursa, have you checked if the thread(s) that are failing are all failing with the same exception?

Reply:
Every time I've seen it fail, it has been one of the two failures that you and Nick reported. One of the directories or the file appears missing.

Reply:
There's no expectation that they will all fail with the same exception. What's happening is that they are all racing on creating the placeholder directories, expanding those directories, then creating the placeholder file, and finally hydrating it. Since we currently have races in our enumeration of directories, one of the directories or the leaf file will sometimes appear to be missing. 

That said, I can definitely update it to print out every exception we get, if you think it'll add more diagnostic value.

Reply:
> That said, I can definitely update it to print out every exception we get, if you think it'll add more diagnostic value.

I think we can start with just printing one.  I agree the most important\interesting part of this test is simply whether or not it succeeds.